Patch - updates maybe_indices_to_slice to use intp_t

benjamindonnachie opened the pull request:
- [ ] closes ##59531 (https://github.com/pandas-dev/pandas/issues/59531)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

benjamindonnachie committed:
Update lib.pyi maybe_indices_to_slice to use uint64

Updates maybe_indices_to_slice to use uint64 allowing massive dataframes to be manipulated (see https://github.com/pandas-dev/pandas/issues/59531)

benjamindonnachie committed:
Update lib.pyx maybe_indices_to_slice to use uint64_t

Update maybe_indices_to_slice to use uint64_t allowing manipulation of massive data frames (See https://github.com/pandas-dev/pandas/issues/59531)

WillAyd reviewed (changes requested):
inp_t is the proper size for an indexer - does it fix things if you try using that consistently?

benjamindonnachie commented:
Happy to give it a try.  I'll let you know.

benjamindonnachie commented:
Do you mean intp_t ?  If so, changing max_len to intp_t and np.intp respectively in lib.pyx and lib.pyi (to match the array definitions) also produces expected results:

```
>>> import pandas as pd
>>> print (pd.__version__)
3.0.0.dev0+1351.g523afa840a.dirty
>>> test = pd.read_pickle("summary.pkl")
>>> summary = test[test['count'] > 0]
>>> print (len(test))
4261028590
>>> print (len(summary))
1621743
```

Very minor changes but happy to submit another PR if that works for you.

benjamindonnachie commented:
Here - try this one https://github.com/pandas-dev/pandas/pull/59538

WillAyd commented:
Yes that's correct. You can just update this PR - no need to create another

benjamindonnachie committed:
Update lib.pyx

Revised to use intp_t for max_len in maybe_indices_to_slice.  As per conversation with WillAyd revised patch as intp_t is the proper size for an indexer.

benjamindonnachie committed:
Update lib.pyi

Updated may_indices_to_slice to use np.intp for max_len.  As per conversation with WillAyd revised patch as intp_t is the proper size for an indexer.

benjamindonnachie commented:
Got there.... Many thanks for your help! :)

WillAyd commented:
@benjamindonnachie great I think this change looks pretty good! Let's see if CI is green.

You will also want to add a whatsnew entry, which can be added to pandas/doc/source/whatsnew/v3.0.0.rst`  You can add it to the section on Bug Fixes > Indexing (just follow the same pattern you see for other notes already there)

benjamindonnachie committed:
Update v3.0.0.rst

Updated whatsnew to reflect fix to maybe_indices_to_slice to address OverflowError.

benjamindonnachie commented:
Whatsnew/v3.0.0.rst updated.  Fingers crossed! :)

mroeschke commented on doc/source/whatsnew/v3.0.0.rst:
```diff
@@ -591,6 +591,7 @@ Indexing
 ^^^^^^^^
 - Bug in :meth:`DataFrame.__getitem__` returning modified columns when called with ``slice`` in Python 3.12 (:issue:`57500`)
 - Bug in :meth:`DataFrame.from_records` throwing a ``ValueError`` when passed an empty list in ``index`` (:issue:`58594`)
+- Bug in :meth:`maybe_indices_to_slice` max_len was set to an int causing ``OverflowError``: value too large to convert to int when manipulating very large dataframes (:issue:`59531`)
```
```suggestion
- Bug in :meth:`DataFrame.__getitem__` when slicing a :class:`DataFrame` many rows raised an ``OverflowError`` (:issue:`59531`)
```

benjamindonnachie committed:
Update doc/source/whatsnew/v3.0.0.rst

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

mroeschke commented:
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

benjamindonnachie commented:
@mroeschke  I'm not sure what else you need given the minor nature of this patch.

mroeschke commented:
It appeared the code checks were not all passing https://github.com/pandas-dev/pandas/actions/runs/10938423968/job/30366432851

swt2c commented:
@benjamindonnachie any plans to revive this PR?  Otherwise, I will - I'm affected by this bug as well.
